### PR TITLE
Proximity Sensor

### PR DESCRIPTION
**Pull request by pvinis**
# Why

https://forums.expo.io/t/proximity-sensor-and-ear-speaker-features/4460

# How

By using the expo barometer and https://github.com/williambout/react-native-proximity as inspiration.

# Test Plan

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/versions/unversioned/sdk/proximity.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+```typescript
+type ProximityMeasurement = {
+  proximityState: boolean;
+};
+```
+
+| Name             | Type                                         | Format     | iOS | Android | Web |
+| ---------------- | -------------------------------------------- | ---------- | --- | ------- | --- |
+| proximityState   | `boolean`                                    | `isClose?` | ✅  | ✅      | ❌  |
+
```
Comment (wodin) [inline docs/pages/versions/unversioned/sdk/proximity.md:161]: I think "near" might be harder to misinterpret than "close".

Comment (pvinis) [reply docs/pages/versions/unversioned/sdk/proximity.md:161]: indeed. I can change it!

File: packages/expo-sensors/ios/EXSensors/EXSensorsManager.m [hunk 2]
```diff
@@ -286,6 +289,38 @@ - (BOOL)isMagnetometerUncalibratedAvailable
...
+{
+  [_proximityHandlers removeObjectForKey:scopedSensorModule];
+  if (_proximityHandlers.count == 0) {
+	[[UIDevice currentDevice] setProximityMonitoringEnabled:NO];
+  }
+}
+
+- (BOOL)isProximityAvailable
+{
+  return YES;
```
Comment (EvanBacon) [inline packages/expo-sensors/ios/EXSensors/EXSensorsManager.m:321]: Not always true: https://developer.apple.com/documentation/uikit/uidevice/1620017-proximitymonitoringenabled need to attempt to enable it and determine if the value is still false.

File: docs/pages/versions/unversioned/sdk/proximity.md (hunk 1 above)
Comment (EvanBacon) [reply docs/pages/versions/unversioned/sdk/proximity.md:161]: Near would also be a bit closer to the W3 proposed spec https://w3c.github.io/proximity/#proximity-sensor-near

**Review by EvanBacon — COMMENTED** [review]
Maybe there should be some naming distinction device proximity and user proximity? Refer to a related spec to ensure this is a scalable API that would make sense across platforms https://whatwebcando.today/proximity.html

**Pull request closed without merging**